src/csharp/CSharpParser.g4

```diff
@@ -377,7 +377,8 @@ query_continuation
 
 //B.2.5 Statements
 statement
-	: labeled_statement			                                     #labeledStatement
+	: ';'                                                            #emptyStatement
```

Comment:
Doesn't this introduce a grammar ambiguity? A single ";" can be resolved as `statement` or as `statement > embedded_statement > embedded_empty_statement`.

Are you sure this line is needed?

Reply:
Antlr resolves to the first matching rule. This rule does shadow the other one, but only in contexts that expect a `statement` node instead of an `embedded_statement`. If I delete this line than all empty statements would get printed as ";".